### PR TITLE
Task/revert to checkout v1

### PR DESCRIPTION
**Pull request by krisorr**
# What it does
* reverts back to checkout v1 api endpoints, as v2 are being deprecated
* updates completeCheckout to do polling internally, and return the completed checkout.  No polling on getCheckoutCompletionStatus is needed now
* updates getShippingRates to poll internally

# Issues
#96 

# Code Review
@sav007 , @yervantk please review

# TBD
Some tests are currently suppressed and depend on #101 before they can be run
More testing of the new polling endpoints, needs to be mocked out so we can force failures

### CODE REVIEW

**Comment by krisorr** [comment]
This PR will need updating in light of #105 to move the polling out of the completeCheckout function and into a convenience function.

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java [hunk 1]
```diff
@@ -153,12 +160,21 @@ public CancellableTask getShippingRates(final String checkoutToken, final Callba
...
-                .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>())
+                .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>(successCodes))
+                .retryWhen(pollingRetryPolicyProvider.provide())
                 .compose(new UnwrapRetrofitBodyTransformer<ShippingRatesWrapper, List<ShippingRate>>())
-                .observeOn(callbackScheduler);
+                .observeOn(callbackScheduler)
+                .doOnError(new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        System.out.println(throwable);
```
Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: We need to introduce SDK logger, as this is not the only place where it's required

Comment (krisorr) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: oops. this was not intentional to leave this in.  I will update.

**Comment by sav007** [comment]
LGTM

**Pull request merged**